cmd/create/cluster/cmd.go

```diff
@@ -2175,38 +2175,37 @@ func getAccountRolePrefix(roleARN string, role aws.AccountRole) (string, error)
 // Validate OpenShift versions
 func validateVersion(version string, versionList []string, channelGroup string, isSTS,
 	isHostedCP bool) (string, error) {
-	if version != "" {
-		// Check and set the cluster version
-		hasVersion := false
-		for _, v := range versionList {
-			if v == version {
-				hasVersion = true
-			}
-		}
-		if !hasVersion {
-			allVersions := strings.Join(versionList, " ")
-			err := fmt.Errorf("A valid version number must be specified\nValid versions: %s", allVersions)
-			return version, err
+	if version == "" {
+		return version, nil
+	}
+	// Check and set the cluster version
+	hasVersion := false
+	for _, v := range versionList {
+		if v == version {
+			hasVersion = true
 		}
+	}
+	if !hasVersion {
+		allVersions := strings.Join(versionList, " ")
+		err := fmt.Errorf("A valid version number must be specified\nValid versions: %s", allVersions)
+		return version, err
+	}
 
-		if isSTS && !ocm.HasSTSSupport(version, channelGroup) {
-			err := fmt.Errorf("Version '%s' is not supported for STS clusters", version)
-			return version, err
+	if isSTS && !ocm.HasSTSSupport(version, channelGroup) {
+		err := fmt.Errorf("Version '%s' is not supported for STS clusters", version)
+		return version, err
+	}
+	if isHostedCP {
+		valid, err := ocm.HasHostedCPSupport(version)
+		if err != nil {
+			return "", fmt.Errorf("error while parsing OCP version '%s': %v", version, err)
 		}
-		if isHostedCP {
-			valid, err := ocm.HasHostedCPSupport(version)
-			if err != nil {
-				return "", fmt.Errorf("error while parsing OCP version '%s': %v", version, err)
-			}
-			if !valid {
-				return "", fmt.Errorf("version '%s' is not supported for hosted clusters", version)
-			}
+		if !valid {
+			return "", fmt.Errorf("version '%s' is not supported for hosted clusters", version)
 		}
-
-		version = "openshift-v" + version
 	}
 
-	return version, nil
+	return ocm.CreateVersionID(version, channelGroup), nil
```

Comment:
CreateVersionID does the prefix of "openshift-v" and adds the suffix for channel group if it is different than stable